tapdb/sqlc/migrations/000013_assets_unique_constraint.up.sql

```diff
@@ -0,0 +1,31 @@
+-- We will apply a new unique constraint on the assets table. This constraint
+-- will be on the columns (anchor_utxo_id, genesis_id, script_key_id).
+-- Rows in the existing table may violate this constraint, so we need to delete
+-- all but one row in each violating set of rows before applying the constraint.
+WITH duplicate_rows AS (
+    SELECT
+        asset_id,
+        anchor_utxo_id,
+        genesis_id,
+        script_key_id,
+
+        -- This is the row number of the row within the set of rows that violate
+        -- the constraint. We will delete all rows with a row number greater
+        -- than 1.
+        ROW_NUMBER() OVER (PARTITION BY anchor_utxo_id, genesis_id, script_key_id ORDER BY asset_id) AS row_num
```

Comment:
Ah, I see now why we'd want to have a way to test migration scripts...
I think this probably works, but we definitely need to make sure it does the correct thing for both Postgres and SQLite.